src/linker/Linker.Steps/BodySubstitutionParser.cs

```diff
@@ -0,0 +1,201 @@
+using System;
+using System.Diagnostics;
+using System.Linq;
+using System.Xml.XPath;
+using Mono.Cecil;
+
+namespace Mono.Linker.Steps
+{
+	public class BodySubstitutionParser : ProcessLinkerXmlStepBase
+	{
+		SubstitutionInfo _substitutionInfo;
+
+		public BodySubstitutionParser (XPathDocument document, string xmlDocumentLocation)
+			: base (document, xmlDocumentLocation)
+		{
+		}
+
+		public BodySubstitutionParser (XPathDocument document, EmbeddedResource resource, AssemblyDefinition resourceAssembly, string xmlDocumentLocation = "")
+			: base (document, resource, resourceAssembly, xmlDocumentLocation)
+		{
+		}
+
+		public SubstitutionInfo Parse (LinkContext context)
```

Comment:
Not important, but I personally would prefer to get the context in the .ctor.

Reply:
Changed